patsy/util.py

```diff
@@ -65,6 +70,11 @@
         )
 have_pandas_categorical_dtype = _pandas_is_categorical_dtype is not None
 
+
+def _pandas_is_string_dtype(x):
```

Comment:
Why do we need this private function? Can't we just consolidate this into `safe_is_pandas_string_dtype`?